pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -267,6 +275,41 @@ func (p *provider) Create(machine *v1alpha1.Machine, _ *cloudprovidertypes.Provi
 		serverCreateOpts.Location = location
 	}
 
+	if c.PlacementGroupPrefix != "" {
+		placementGroups, _, err := client.PlacementGroup.List(ctx, hcloud.PlacementGroupListOpts{Type: hcloud.PlacementGroupTypeSpread})
+		if err != nil {
+			return nil, hzErrorToTerminalError(err, "failed to get placement groups of type spread")
+		}
+		var n *hcloud.PlacementGroup
+		for _, p := range placementGroups {
+			if !strings.HasPrefix(p.Name, c.PlacementGroupPrefix) {
+				continue
+			}
+			if len(p.Servers) < 10 {
+				n = p
+				break
+			}
+		}
+		if n == nil {
+			labels := map[string]string{}
+			for k, v := range c.Labels {
+				if k != machineUIDLabelKey {
+					labels[k] = v
+				}
+			}
+			p, _, err := client.PlacementGroup.Create(ctx, hcloud.PlacementGroupCreateOpts{
+				Name:   fmt.Sprintf("%s-%s", c.PlacementGroupPrefix, rand.SafeEncodeString(rand.String(5))),
+				Labels: labels,
+				Type:   hcloud.PlacementGroupTypeSpread,
+			})
+			if err != nil {
+				return nil, hzErrorToTerminalError(err, "failed to create placement group")
+			}
+			n = p.PlacementGroup
+		}
+		serverCreateOpts.PlacementGroup = n
+	}
+
```

Comment:
There are bunch of one letter variables with places of use far then 1-3 lines after declaration. Please give them proper names, it's hard to read (I know, we use them too and I'm not proud of it, but it's legacy, at least let's not increase that problem in the new code)

Reply:
hehe ok, I thought the short variables are your coding style 😄